pkg/prometheus/server/operator.go

```diff
@@ -964,6 +972,46 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+// add or remove finalizers for the Prometheus resource.
+// Returns true if the finalizers were modified, otherwise false. The second return value is an error, if any.
+func (c *Operator) syncFinalizers(ctx context.Context, p *monitoringv1.Prometheus, key string) (bool, error) {
+	if !c.configResourcesStatusEnabled {
+		return false, nil
+	}
+	finalizersChanged := false
+	if !c.rr.DeletionInProgress(p) {
+		// Add finalizer to the Prometheus resource if it doesn't have one.
+		finalizers := p.GetFinalizers()
+		patchBytes, err := k8sutil.FinalizerAddPatch(finalizers, k8sutil.StatusCleanupFinalizerName)
+		if err != nil {
+			return finalizersChanged, fmt.Errorf("failed to marshal patch: %w", err)
+		}
+		if len(patchBytes) > 0 {
```

Comment:
(nit) It reduces the number of nested levels.
```suggestion
		if len(patchBytes) == 0 {
			return false, nil
		}
```